added imaging tutorial

ThomasWollmann opened the pull request.

shiltemann commented on topics/imaging/tutorials/imaging-introduction/tutorial.md:
```diff
@@ -0,0 +1,163 @@
+> 1. **Image Info** {% icon tool %} with the following parameters to extract metadata from the image:
+>    - {% icon param-file %} *"Input Image"*: `input.tif` file (output of the previous step)
+> 2. Click on the {% icon galaxy-eye %} (eye) icon next to the file name, to look at the file content and search for image acquisition information
+>    ![eye](../../images/eye-icon.png){:width="320px"}
```
we try to avoid screenshots as much as possible, this probably isn't needed? the text description is probably enough? If you really feel strongly about including it, could you turn it into a snippet so that if the UI changes in the future we only have to change it in one place?

shiltemann commented on topics/imaging/tutorials/imaging-introduction/tutorial.md:
```diff
@@ -0,0 +1,163 @@
+>    ![eye](../../images/eye-icon.png){:width="320px"}
+{: .hands_on}
+
+# Image Convertion
```
```suggestion
# Image Conversion
```

shiltemann commented on topics/imaging/tutorials/imaging-introduction/tutorial.md:
```diff
@@ -0,0 +1,163 @@
+
+# Getting data
+
+The dataset required for this tutorial contains a screen of DAPI stained HeLa nuclei ([more information](https://zenodo.org/record/3358874#.XUQ9oi1XZUM)). We will use a sample image from this dataset for training basic imaging skills in Galaxy.
```
Could you give a bit more of an introduction/motivation? what are these images? what is the research/biological question we are trying to answer in this tutorial?

ThomasWollmann replied:
This tutorial is just introductory to image handling there is no research/biological question.

shiltemann replied:
it doesn't have to be much, even if it is just "we have an image of cells and want to detect the nuclei and segment the image"

just a thought, is it possible to add a step at the end that counts the number of cells? because that would be a neat little result maybe (but no worries if not possible)

shiltemann commented on topics/imaging/tutorials/imaging-introduction/tutorial.md:
```diff
@@ -0,0 +1,163 @@
+> 1. **Convert image** {% icon tool %} with the following parameters to convert the image to PNG:
+>    - {% icon param-file %} *"Input Image"*: `input.tif` file (output of the previous step)
+>    - *"Output data type"*: `PNG`
+> 2. Rename the generated file to `viz_input`
```
you could add `{% include snippets/rename_dataset.md %}` on a line below this to remind people how to rename a dataset?

shiltemann replied:
oh, I see now that you already did this in an earlier box, that is probably good enough :)

ThomasWollmann replied:
@shiltemann I tried to address your comments in the last commit.

shiltemann commented:
I just ran through the tutorial, it was fun, thanks!

- I would maybe add some of the png image outputs to the manual, they are cool, and if people don't have time to finish or if they mess up, then they can at least see the output they were supposed to get, what do you think?

- The input dataset is large (500Mb) and we only use one image (2.6Mb) from it, could we just upload the single file to zenodo? or if unzipping is something you want to teach them, maybe just create a zip file with only that image and possibly one or two others? it will save some time on import and some disk space on the server

- Please still add a workflow file btw (since I just performed the tutorial I could extract it from my history too if you prefer)

again, very cool, thanks a lot!

ThomasWollmann commented:
> I just ran through the tutorial, it was fun, thanks!
> 
> * I would maybe add some of the png image outputs to the manual, they are cool, and if people don't have time to finish or if they mess up, then they can at least see the output they were supposed to get, what do you think?
Yes this would be nice! Since, you already have the result images you can maybe just upload them :)
> * The input dataset is large (500Mb) and we only use one image (2.6Mb) from it, could we just upload the single file to zenodo? or if unzipping is something you want to teach them, maybe just create a zip file with only that image and possibly one or two others? it will save some time on import and some disk space on the server
Soon, I will add a second tutorial next, which is using the whole dataset and should be teached along with this tutorial. Therefore, I think we can just keep the large dataset. Also makes sense to teach people from the beginning that imaging datasets are often not just single files.
> * Please still add a workflow file btw (since I just performed the tutorial I could extract it from my history too if you prefer)
Would be nice if you can add it, since you have it already. Thank you in advance!
> 
> again, very cool, thanks a lot!

shiltemann commented:
@ThomasWollmann ok, no problem, I will add the images and workflow on Monday :) 

and cool about the workshop! have a great weekend!

shiltemann commented:
Hi @ThomasWollmann I added the images and workflow, could you please check if it looks ok to you?

In the final image, I get a yellow label of `65535` on the image, is that meant to happen or did I do something wrong somewhere? and if not, maybe explain what this label signifies?

shiltemann commented on topics/imaging/tutorials/imaging-introduction/tutorial.md:
```diff
@@ -0,0 +1,206 @@
+>    - {% icon param-file %} *"Input Image"*: `input_smoothed_normalized` file (output of **Histogram equalization** {% icon tool %})
+>    - *"Output data type"*: `PNG`
+> 6. Rename {% icon galaxy-pencil %} the generated file to `viz_smoothed_normalized`
+> 7. Click on the {% icon galaxy-eye %} (eye) icon next to the file name, to look at the file content and compare the result with `viz_normalized`.
```
here you ask them to compare the two images, could you maybe write a sentence about what they should observe (maybe under the image below)?

ThomasWollmann commented:
> Hi @ThomasWollmann I added the images and workflow, could you please check if it looks ok to you?
> 
> In the final image, I get a yellow label of `65535` on the image, is that meant to happen or did I do something wrong somewhere? and if not, maybe explain what this label signifies?

Yes, I missed a step. I'll add it to the tutorial and then you can try again ;)

ThomasWollmann commented:
@shiltemann I changed the description. Can you please try it again and update the results?

shiltemann commented on topics/imaging/tutorials/imaging-introduction/tutorial.md:
```diff
@@ -178,19 +178,22 @@ Objects of interest like nuclei can be segmented by using a smoothed image and t
+> 3. **Binary 2 Label** {% icon tool %} with the following parameters to segment the image:
+>    - {% icon param-file %} *"Binary Image File"*: `input_segmented` file (output of **Auto Threshold** {% icon tool %})
+> 4. Rename {% icon galaxy-pencil %} the generated file to `input_segmented_labeled`
+> 5. **Convert image** {% icon tool %} with the following parameters to convert the image to PNG:
```
@ThomasWollmann was just testing this to extract the updated workflow, but this image now looks completely black, is that expected?

ThomasWollmann replied:
Guess this is the desired output. However, it is not nicely visible due to the low label ids. Try running histogram normalization before converting it. If this works, we can add it to the tutorial.

shiltemann replied:
@ThomasWollmann ah yes, that's nicer :) I was thinking, what if we also do a `Count objects` step at the end, that would be a nice little result/objective maybe "here is an image, how many cells does it contain"? what do you think?

ThomasWollmann replied:
@shiltemann Nice idea. Feel free to add the result, since you already have the files in history. I'll add the text except the number.

ThomasWollmann commented:
@shiltemann I added the extra step and some questions regarding this issue to the tutorial. Do you think 30 minutes is too optimistic for this tutorial?

ThomasWollmann commented:
@shiltemann Currently, I'm preparing the second tutorial. Should I push it to the same PR?

shiltemann commented:
@ThomasWollmann whatever you prefer. If you're ok with this staying open a little longer then go ahead and add it here :)

ThomasWollmann commented:
> @ThomasWollmann whatever you prefer. If you're ok with this staying open a little longer then go ahead and add it here :)

Since you're already into the topic, we can just process both at once ;)

ThomasWollmann commented:
@shiltemann Now, I added the the second tutorial. It closer to a real use case. I know that it still lacks some biological motivation, which I'll add later if your a satisfied with by technical description.

ThomasWollmann commented:
> @shiltemann Now, I added the the second tutorial. It closer to a real use case. I know that it still lacks some biological motivation, which I'll add later if your are satisfied with by technical description.

ThomasWollmann commented:
@shiltemann I talked to the guy who imaged the dataset. I think I can change the second tutorial to make it closer to a real use case and also reduce the dataset size. Just give me some time to adapt the tutorial ;)

ThomasWollmann commented:
@shiltemann I changed the dataset and the description. I think the second tutorial is now more useful for practitioners. Please take a look!

ThomasWollmann commented:
@bgruening Now, the splitted dataset is small and only two subsets are required for the tutorials. This should solve the issues for EU?

bgruening commented:
> @bgruening Now, the splitted dataset is small and only two subsets are required for the tutorials. This should solve the issues for EU?

Did you had issues? I could run the other completely. I would like to keep the old datasets and workflow for testing. I think its a cool example where we can benchmark the system.

ThomasWollmann commented:
> > @bgruening Now, the splitted dataset is small and only two subsets are required for the tutorials. This should solve the issues for EU?
> 
> Did you had issues? I could run the other completely. I would like to keep the old datasets and workflow for testing. I think its a cool example where we can benchmark the system.

I just took time. Now I splitted the dataset into the different treatments and in the course you can actually compare phenotypes visually or apply machine learning on the extracted features.

The full dataset is still on Zenodo. I think it is a good example for preparing Galaxy for large scale imaging datasets. Keep in mind that this dataset is still very small ;)

shiltemann commented:
thanks a lot @ThomasWollmann :) The second tutorial looks good, I like that they also learn about workflows a bit at the same time :)

I reworked the last hands_on section of the first tutorial a bit, because using the normalized image resulted in higher IDs that overlapped in the final image and didnt look as nice

![image](https://user-images.githubusercontent.com/2563865/62783314-6c978c80-babc-11e9-99ea-06cd18ebcd04.png)

so please check that it all still looks ok to you :)

ThomasWollmann commented:
> thanks a lot @ThomasWollmann :) The second tutorial looks good, I like that they also learn about workflows a bit at the same time :)
> 
> I reworked the last hands_on section of the first tutorial a bit, because using the normalized image resulted in higher IDs that overlapped in the final image and didnt look as nice
> 
> ![image](https://user-images.githubusercontent.com/2563865/62783314-6c978c80-babc-11e9-99ea-06cd18ebcd04.png)
> 
> so please check that it all still looks ok to you :)

Hmm since the image contains many objects, we probably should not display the ids. In the overlay tool you can set "Plot Labels" to "No". I think this looks much more useful. @shiltemann What do you think?

@shiltemann Thank you for putting so much effort in improving the quality of the tutorials!

shiltemann commented:
oh, that was just how it looked before, to explain why I changed it, without the histogram equalization step it looks quite ok now:

![image](https://user-images.githubusercontent.com/2563865/62788284-de74d380-bac6-11e9-96f7-fe9a2498a919.png)

So I just reverted to your original steps, and made the histogram equalization step an exercise to help explain the completely black image they get, but use the original image for the segmentation mask overlay. What do you think?

ThomasWollmann commented:
> oh, that was just how it looked before, to explain why I changed it back. ..without the histogram equalization step it looks quite ok now:
> 
> ![image](https://user-images.githubusercontent.com/2563865/62788284-de74d380-bac6-11e9-96f7-fe9a2498a919.png)
> 
> So I just reverted to your original steps, and made the histogram equalization step an exercise to help explain the completely black looking image they get, but use the original image for the segmentation mask overlay. What do you think?

Sounds good! Now everything looks fine :)

ThomasWollmann commented:
@shiltemann @bgruening still something missing for this tutorial?

bgruening commented on topics/imaging/tutorials/hela-screen-analysis/tutorial.md:
````diff
@@ -95,7 +101,7 @@ The dataset required for this tutorial contains a screen of DAPI stained HeLa nu
 > 9. Upload {% icon galaxy-upload %} the following segmentation filter rules as a new pasted file (format: `tabular`):
 >    ```
->    area	eccentricity
+>    	area	eccentricity
````
that looks wrong here. the first one

ThomasWollmann replied:
Why? The file has three columns. The first column has no heading.

bgruening replied:
You are correct. I have seen two ... but it seems to be just one.

:+1: from me.

shiltemann commented:
> @shiltemann @bgruening still something missing for this tutorial?

Hi @ThomasWollmann oh, you mentioned you were planning to add more biological motivation so I thought you were still working on this

Could you just add some `key_points` to the tutorial metadata? just the main conclusions/take home messages you want people to remember from the tutorials.

Also: do we need to cite the person who created the dataset somehow? was it part of a publication or project we can refer to?

ThomasWollmann commented:
@shiltemann The data was created by Manuel Gunkel as part of de.NBI Systematic Phenotyping. I do not think that we need additional citations, since we have the zenodo record.

ThomasWollmann commented:
@shiltemann Now, I added some more motivation and key_points ;)

shiltemann commented:
@ThomasWollmann thanks! This looks great!

ThomasWollmann commented:
Wow, @shiltemann awesome work. Thank you for the support!

shiltemann commented:
:tada:
